### PR TITLE
Test class

### PR DESCRIPTION
**Pull request by ferricles**
No longer has a test class; name is a misnomer.

Provides basic test functionality via multiple flavors of assert.

### CODE REVIEW

File: js/utils.js [hunk 1]
```diff
@@ -412,4 +412,114 @@ async function hardReload() {
...
+    script.onerror = reject
+  
+    script.onload = script.onreadystatechange = function() {
+      const loadState = this.readyState
+  
+      if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
+  
+      script.onload = script.onreadystatechange = null
+  
+      resolve()
```
Comment (hppeng-wynn) [inline js/utils.js:434]: semicolons

File: js/utils.js [hunk 2]
```diff
@@ -412,4 +412,114 @@ async function hardReload() {
...
+
+/** Asserts proximity between 2 arguments. Should be used for any floating point datatype.
+ * 
+ * @param {*} arg1 - first argument to compare.
+ * @param {*} arg2 - second argument to compare.
+ * @param {Number} epsilon - the margin of error (<= del difference is ok).
+ * @param {String} msg - the error message to throw. 
+ */
+function assert_near(arg1, arg2, epsilon, msg) {
+    if (Math.abs(arg1 - arg2) > epsilon) {
```
Comment (hppeng-wynn) [inline js/utils.js:486]: possibly give a default eps

File: js/utils.js [hunk 3]
```diff
@@ -412,4 +412,114 @@ async function hardReload() {
...
+
+/** Asserts that there is an error when a callback function is run.
+ * 
+ * @param {Function} func_binding - a function binding to run. Can be passed in with func.bind(null, arg1, ..., argn)
+ * @param {String} msg - the error message to throw.
+ */
+function assert_error(func_binding, msg) {
+    try {
+        func_binding();
+        console.trace(msg ? msg : "Function didn't throw an error.");
```
Comment (hppeng-wynn) [inline js/utils.js:521]: are you trying to throw an error like the other ones? why not throw error outside the try catch

Comment (ferricles) [reply js/utils.js:521]: i didn't think about it lmao

**Review by hppeng-wynn — APPROVED** [review]
lgtm

**Pull request merged**